vault/identity_store_util.go

```diff
@@ -50,6 +50,9 @@ func (c *Core) loadIdentityStoreArtifacts(ctx context.Context) error {
 		if err := c.identityStore.loadCachedEntitiesOfLocalAliases(ctx); err != nil {
 			return err
 		}
+		if err := c.identityStore.storeOIDCDefaultResources(ctx); err != nil {
```

Comment:
I believe this code will run on Performance standby nodes and the Active node on a Performance Secondary Cluster. Both of which will be unable to write this data locally. I think we need to only call this function of we are the active node on a performance primary cluster?

Reply:
I think we just need to skip on `if c.perfStandby {}` right?  For handling peformance secondary on the active node, you might have to move the call into this block: https://github.com/hashicorp/vault/blob/2fe214f5719a17b67f2630f82f8e51797db1a633/vault/core.go#L2138

Reply:
The problem I saw was that `c.perfStandby` was always true when this part of the code is called. I logged the value (and used debugger) to confirm this with a 3 instance cluster. This resulted in none of the default resources being written.

Reply:
> The problem I saw was that c.perfStandby was always true when this part of the code is called

🤔 I don't think this is supposed to be the case. `c.perfStandby` should only be true if this function call came from the enterprise `perfStandbyUnsealStrategy.unseal` code path. 

I believe something like this should work:

```
if !c.perfStandby && c.isPrimary() { 
    if err := c.identityStore.storeOIDCDefaultResources(ctx); err != nil { 
        return err 
    } 
}
```

Reply:
I think that Austin was hitting this issue with a 3-node Vault ent cluster that has perf standby enabled. I do see c.perfStandby being set to true, and then followed by a call do to c.postUnseal which eventually calls this. The additional check that you have on `c.isPrimary()` would make it work though.

Reply:
Thanks for the info. I'll give that a shot really quick.

Out of curiosity, is checking `!c.perfStandby` via the result of [c.HAState()](https://github.com/hashicorp/vault-enterprise/blob/f684c09f39b7d99c0237534bf5a29f7577455c37/vault/core.go#L578-L587) any different from what I originally tried? 🤔

Reply:
Ah, disregard my question above. I think I understand how that's set now.

Reply:
@austingebauer You wrote above that you originally tried `c.HAState() == consts.Active` which i think would always be false at that point in time. I think if you did `c.HAState() != consts.PerfStandby` it may work work though!

Reply:
@austingebauer Yes, i think that's correct

Reply:
Thanks for the help, both! TIL as well. Testing this really quick then will push up the changes.